### PR TITLE
Add SCT from future check

### PR DESCRIPTION
**Pull request by Mercurrent**

### CODE REVIEW

File: certsubmitter/certsubmitter.go [hunk 1]
```diff
@@ -124,6 +124,12 @@ func checkSCT(sct *ct.SignedCertificateTimestamp, chain []*x509.Certificate, sv
 		errs = append(errs, &errors.SignatureVerificationError{Err: err})
 	}
 
+	// Verify SCT timestamp.
```
Comment (NatalieDoduc) [inline certsubmitter/certsubmitter.go]: nit, stuttering: "Signed Certificate Timestamp" timestamp. I don't see a way around this, and I also understand that's how the struct is built and named.

File: certsubmitter/certsubmitter.go [hunk 2]
```diff
@@ -124,6 +124,12 @@ func checkSCT(sct *ct.SignedCertificateTimestamp, chain []*x509.Certificate, sv
...
+	sctTimestamp := ct.TimestampToTime(sct.Timestamp)
+	if receivedAt.Sub(sctTimestamp).Seconds() < 0 {
```
Comment (NatalieDoduc) [inline certsubmitter/certsubmitter.go]: Use After() instead?

Comment (Mercurrent) [reply certsubmitter/certsubmitter.go]: My bad. Current monitor forgives gaps of up to 1 second. Let's stick to it for proper functionality alignment.

File: certsubmitter/certsubmitter.go (hunk 1 above)
Comment (Mercurrent) [reply certsubmitter/certsubmitter.go]: oh well. Naming is not our strongest skill.

**Comment by Mercurrent** [comment]
Comments addressed. Time gap moved to milliseconds.

File: certsubmitter/certsubmitter.go [hunk 3]
```diff
@@ -124,6 +124,13 @@ func checkSCT(sct *ct.SignedCertificateTimestamp, chain []*x509.Certificate, sv
 		errs = append(errs, &errors.SignatureVerificationError{Err: err})
 	}
 
+	// Verify SCT timestamp.
+	sctTimestamp := ct.TimestampToTime(sct.Timestamp)
+	// Consider only gaps bigger than 1 second
```
Comment (NatalieDoduc) [inline certsubmitter/certsubmitter.go]: nit: How about "Consider gaps bigger than 1 second as errors." Or the inverse "We accept timestamps up to 1 second after reception time by XXX." - ideally specifying who XXX that received the data is.

Would sctTimestamp.Sub(receivedAt).Milliseconds() < 1000 be a bit more readable?

Comment (Mercurrent) [reply certsubmitter/certsubmitter.go]: Adjusted.

**Pull request merged**